gammapy/data/event_list.py

```diff
@@ -386,13 +386,114 @@ def _get_lon_lat(self, header):
-        self.plot_energy_dependence(ax=axes[1])
-        self.plot_offset_dependence(ax=axes[2])
+        self.plot_time_map(ax=axes[1])
+
+        #log-log scale for time map
+        xlims = axes[1].set_xlim()
+        ylims = axes[1].set_ylim()
+        axes[1].set_xlim(1e-3, xlims[1])
+        axes[1].set_ylim(1e-3, ylims[1])
+        axes[1].loglog()
+        # TODO: self.plot_energy_dependence(ax=axes[x])
+        # TODO: self.plot_offset_dependence(ax=axes[x])
         plt.tight_layout()
         plt.show()
 
+    def plot_image(self, ax= None, number_bins=50):
+        """Plot the counts as a function of x and y camera coordinate."""
+        import matplotlib.pyplot as plt
+        ax = plt.gca() if ax is None else ax
+
+        max_x = max((self[:]['DETX']))
+        min_x = min((self[:]['DETX']))
+        max_y = max((self[:]['DETY']))
+        min_y = min((self[:]['DETY']))
+
+        x_edges = np.linspace(min_x, max_x, number_bins)
+        y_edges = np.linspace(min_y, max_y, number_bins)
+
+        count_image, x_edges, y_edges = np.histogram2d(self[:]['DETY'], self[:]['DETX'], bins=(x_edges, y_edges))
+
+        ax.set_title('# Photons')
+
+        ax.imshow(count_image, interpolation='nearest', origin='low',
+                  extent=[x_edges[0], x_edges[-1], y_edges[0], y_edges[-1]])
+
+    def plot_energy_dependence(self, ax = None):
+        import matplotlib.pyplot as plt
+        ax = plt.gca() if ax is None else ax
+
+    def plot_offset_dependence(self, ax = None):
+        import matplotlib.pyplot as plt
+        ax = plt.gca() if ax is None else ax
+
+    def plot_time_map(self, ax=None):
+        """
+        A time map showing for each event the time between the previous and following event.
+
+        The use and implementation are described here
+
+        https://districtdatalabs.silvrback.com/time-maps-visualizing-discrete-events-across-many-timescales
+
+        Parameters
+        ----------
+
+        ax : `~matplotlib.axes.Axes` or None
+            Axes
+
+        Returns
+        -------
+        ax : `~matplotlib.axes.Axes`
+            Axes
+
+        Examples
+        --------
+        Plot a time map of the events:
+
+        .. plot::
+            :include-source:
+
+        from gammapy.data import EventList, EventListDataset, EventListDatasetChecker
+        from gammapy.datasets import gammapy_extra
+
+        import os
+        import numpy
+        import matplotlib.pyplot as plt
+
+        os.environ["GAMMAPY_EXTRA"] = "/home/jon/Pycharm/gammapy-extra"
+        filename = gammapy_extra.filename('datasets/hess-crab4/hess_events_simulated_023523.fits')
+        event_list = EventList.read(filename, hdu='EVENTS')
+
+        event_list.plot_time_map()
+        
+        plt.show()
+
+        """
+
+        import matplotlib.pyplot as plt
+
+        ax = plt.gca() if ax is None else ax
+
+        first_event_time = np.min(self[:]['TIME'])
+
+        # Note the events are not necessarily in time order
```

Comment:
Pretty sure the example there was not in time order....
It might have been me doing something silly in IPython, but I did run into a problem with it.  Unless you're worried about performance, I would rather leave it in as defensive programming.